payu/models/access.py

```diff
@@ -89,61 +100,68 @@ def setup(self):
             if model.model_type in ('cice', 'matm'):
 
                 # Update the supplemental OASIS namelists
-                cpl_fname, cpl_group, runtime0_key = cpl_keys[model.model_type]
 
-                cpl_fpath = os.path.join(model.work_path, cpl_fname)
+                # cpl_nml is the coupling namelist copied from the control to
+                # work directory.
+                cpl_fpath = os.path.join(model.work_path, model.cpl_fname)
                 cpl_nml = f90nml.read(cpl_fpath)
 
                 # Which calendar are we using, noleap or Gregorian.
-                caltype = cpl_nml[cpl_group]['caltype']
-                init_date = cal.int_to_date(cpl_nml[cpl_group]['init_date'])
+                caltype = cpl_nml[model.cpl_group]['caltype']
```

Comment:
```suggestion
                # Use variable to point directly to coupling namelist group
                cpl_group = cpl_nml[model.cpl_group]
                caltype = cpl_group['caltype']
```
If you want to can make a variable to point directly to the coupling namelist group, which would clean up the later sections which also set variables here. It's a pattern that used elsewhere in the code.

Just a suggestion, up to you if you think it improves or degrades readability.

Reply:
Great idea! I think it makes it easier to read, and added this change in to the latest commit.